OpenRA.Game/FileSystem/ZipFile.cs

```diff
@@ -206,7 +206,9 @@ public Stream GetSource()
 
 		public bool TryParsePackage(Stream s, string filename, FileSystem context, out IReadOnlyPackage package)
 		{
-			if (!Extensions.Any(e => filename.EndsWith(e, StringComparison.InvariantCultureIgnoreCase)))
+			var signature = s.ReadUInt32();
+
+			if (signature != Signature)
```

Comment:
This doesn't read well, could the '**S**ignature' variable be more descriptive to signify why it's different to **s**ignature here?